src/main/java/org/folio/service/transactions/batch/AbstractBatchTransactionService.java

```diff
@@ -0,0 +1,93 @@
+package org.folio.service.transactions.batch;
+
+import io.vertx.core.json.JsonObject;
+import org.folio.rest.exception.HttpException;
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.Transaction;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.lang.Boolean.TRUE;
+import static java.util.Objects.requireNonNullElse;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.CREDIT;
+
+public abstract class AbstractBatchTransactionService implements BatchTransactionServiceInterface {
+
+  Map<Budget, List<Transaction>> createBudgetMapForTransactions(List<Transaction> transactions, List<Budget> budgets) {
+    return transactions.stream()
+      .collect(groupingBy(tr -> budgetForTransaction(budgets, tr)))
+      .entrySet()
+      .stream()
+      .filter(entry -> entry.getKey().isPresent())
+      .collect(toMap(entry -> entry.getKey().get(), Map.Entry::getValue));
+  }
+
+  Optional<Budget> budgetForTransaction(List<Budget> budgets, Transaction tr) {
```

Comment:
better to name like a verb: for example getBudgetForTransaction()

Reply:
OK